File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/build_tools/maven/MavenSettings.java
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.build_tools.maven;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableMavenSettings.class)
public interface MavenSettings {
  Optional<MavenImportingSettings> importing();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/build_tools/maven/MavenSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.build_tools.maven

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.File

data class MavenSettings(

    @JsonProperty("Importing")
    val importing: MavenImportingSettings? = null,

    @JsonProperty("Maven home directory")
    val mavenHomeDirectory: File? = null,

    @JsonProperty("Thread count")
    val threadCount: String? = null
)
